classes/author/Collector.inc.php

```diff
@@ -0,0 +1,232 @@
+    {
+        $this->givenName = $givenName;
+        $this->familyName = $familyName;
+        return $this;
+    }
+
+    /**
+     * Filter by the specified country code
+     *
+     * @param $country string Country code (2-letter)
+     *
+     * */
+    public function filterByCountry(string $country): self
+    {
+        $this->country = $country;
+        return $this;
+    }
+
+    /**
+     * Filter by the specified affiliation code
+     *
+     * */
+    public function filterByAffiliation(string $affiliation): self
+    {
+        $this->affiliation = $affiliation;
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     */
+    public function limit(?int $count): self
+    {
+        $this->count = $count;
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     */
+    public function offset(?int $offset): self
+    {
+        $this->offset = $offset;
+        return $this;
+    }
+
+    /**
+     * @copydoc CollectorInterface::getQueryBuilder()
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $q = DB::table('authors as a')
+            ->Join('publications as p', 'a.publication_id', '=', 'p.publication_id')
+            ->Join('submissions as s', 'p.submission_id', '=', 's.submission_id');
+
+        if (isset($this->contextIds)) {
+            $q->whereIn('s.context_id', $this->contextIds);
+        }
+
+        if (isset($this->familyName) && isset($this->givenName)) {
+            $familyName = $this->familyName;
+            $givenName = $this->givenName;
+            $q->leftJoin('author_settings as fn', 'a.author_id', '=', 'fn.author_id')
+                ->where(function ($q) use ($familyName) {
+                    $q->where('fn.setting_name', '=', 'familyName');
+                    $q->where('fn.setting_value', '=', $familyName);
+                });
+            $q->leftJoin('author_settings as gn', 'a.author_id', '=', 'gn.author_id')
+                ->where(function ($q) use ($givenName) {
+                    $q->where('gn.setting_name', '=', 'givenName');
+                    $q->where('gn.setting_value', '=', $givenName);
+                });
+        }
+
+        if (isset($this->publicationIds)) {
+            $q->whereIn('a.publication_id', $this->publicationIds);
+        }
+
+        if (isset($this->country)) {
+            $country = $this->country;
+            $q->join('author_settings as cs', 'a.author_id', '=', 'cs.author_id')
+                ->where(function ($q) use ($country) {
+                    $q->where('cs.setting_name', '=', 'country');
+                    $q->where('cs.setting_value', '=', $country);
+                });
+        }
+
+        if (isset($this->affiliation)) {
+            $affiliation = $this->affiliation;
+            $q->join('author_settings as afs', 'a.author_id', '=', 'afs.author_id')
```

Comment:
Because there might be multiple affiliations for each author (multilingual), and several affiliations might match the query, you'd need to make this a `whereIn` with subquery to avoid including the same author multiple times.

Reply:
Don't forget this one!

Reply:
Fixed (I think) also with other attributes